metsim/metsim.py

```diff
@@ -42,71 +47,143 @@ def __init__(self, params:dict):
         """
         # Record parameters
         self.update(params)
+        MetSim.params['dates'] = pd.date_range(params['start'], params['stop'])
+        self.output = None
 
         # Get the necessary information from the domain
-        domain = Dataset(MetSim.params['domain'], 'r')
-        self.elev = np.array(domain['elev'])
-        self.lats = np.array(domain['lat'])
-        self.lons = np.array(domain['lon'])
-        domain.close()
-
-        # If we are outputting netcdf, intiialize the file
-        if MetSim.params.get('out_format', 'netcdf') == 'netcdf':
-            self.init_netcdf()
-
-
-    def run(self, forcings):
+        self.domain = Dataset(MetSim.params['domain'], 'r')
+        self.domain_lats = np.array(self.domain['lat'])
+        self.domain_lons = np.array(self.domain['lon'])
+        self.elev = np.array(self.domain['elev'])
+
+        # Create the data structures for writeout
+        self.manager = Manager()
+        self.queue = self.manager.Queue()
+
+
+    def launch(self, job_list):
+        """Farm out the jobs to separate processes"""
+        # Used to dispatch to the correct preprocessing functions
+        in_preprocess = {"ascii" : self.ascii_in_preprocess,
+                         "binary" : self.binary_in_preprocess,
+                         "netcdf" : self.netdf_in_preprocess}
+        met_data = in_preprocess[MetSim.params['in_format']](job_list)
+
+        # Start up the IO Process
+        io_process = Process(target=self.create_io_process, args=(self.queue,))
+        io_process.start()
+
+        # Split the jobs up into groups based on the number
+        # of desired processes
+        self.locations = np.array_split(np.array(self.locations),  MetSim.params['nprocs'])
```

Comment:
Filter the locations for active grid cells before farming them out. This should get rid of the need for the `try: ... except: ...` statements around the interpolator deeper in the code and it makes sure each process has the same number of *active* cells